### PR TITLE
REF: de-nest Series getitem/setitem checks

### PR DESCRIPTION
**Pull request by jbrockmendel**
ATM the first checks we do in `Series.__getitem__` and `__setitem__` are for `get_loc` with some not-totally-straightforward try/except logic.

This moves the slice case and the com.is_bool_indexer case up, and aligns the ordering of the checks between getitem/setitem.

Getting the com.is_bool_indexer case out from the try/except in setitem is particularly nice.

That said, this _may_ not be what we want to do.  This will very slightly improve performance in the cases moved up, and equally slightly harm performance in the scalar case.  AFAIK the current ordering is not based on optimizing one case over the others.

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -897,26 +908,14 @@ def __getitem__(self, key):
                 else:
                     raise
 
-        if not key_is_scalar:
-            # avoid expensive checks if we know we have a scalar
```
Comment (jorisvandenbossche) [inline pandas/core/series.py:901]: This comment might be relevant? Also if you move it up as you did now, you can keep this scalar check?

**Review by jorisvandenbossche — COMMENTED** [review]
I suppose the `isinstance(key, slice)` is not costly, but about the `is_bool_indexer`, I am not so sure. I would at least do some timings to check that (see also the comments that were present)

**Comment by jbrockmendel** [comment]
> but about the is_bool_indexer, I am not so sure

yah, im torn on this.  I think following #33404 i can improve the try/except blocks to allow us to retain the current ordering while also getting the de-nesting we get here

**Comment by jreback** [comment]
can you rebase following #33404 merge with any changes you are planning

**Comment by jbrockmendel** [comment]
mothballing until after the next pass

**Pull request closed without merging**